odoo/addons/openupgrade_records/models/analysis_wizard.py

```diff
@@ -0,0 +1,141 @@
+# coding: utf-8
+# Copyright 2011-2015 Therp BV <https://therp.nl>
+# Copyright 2016 Opener B.V. <https://opener.am>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import os
+from odoo.modules import get_module_path
+
+from odoo import api, fields, models
+from odoo.addons.openupgrade_records.lib import compare
+
+
+class AnalysisWizard(models.TransientModel):
+    _name = 'openupgrade.analysis.wizard'
+    _description = 'OpenUpgrade Analysis Wizard'
+    server_config = fields.Many2one(
+        'openupgrade.comparison.config',
+        'Configuration', required=True)
+    state = fields.Selection(
+            [('init', 'Init'), ('ready', 'Ready')],
+            readonly=True, default='init')
+    log = fields.Text()
+    write_files = fields.Boolean(
+        help='Write analysis files to the module directories',
+        default=True)
+
+    @api.multi
+    def get_communication(self):
+        """
+        Retrieve both sets of database representations,
+        perform the comparison and register the resulting
+        change set
+        """
+
+        def write_file(module, version, contents,
+                       filename='openupgrade_analysis.txt'):
+            module_path = get_module_path(module)
+            if not module_path:
+                return "ERROR: could not find module path:\n"
+            full_path = os.path.join(
+                module_path, 'migrations', version)
+            if not os.path.exists(full_path):
+                try:
+                    os.makedirs(full_path)
+                except os.error:
+                    return "ERROR: could not create migrations directory:\n"
+            logfile = os.path.join(full_path, filename)
+            try:
+                f = open(logfile, 'w')
+            except Exception:
+                return "ERROR: could not open file %s for writing:\n" % logfile
+            f.write(contents)
+            f.close()
+            return None
+
+        self.ensure_one()
+        connection = self.server_config.get_connection()
+        remote_record_obj = connection.get_model('openupgrade.record')
+        local_record_obj = self.env['openupgrade.record']
+
+        # Retrieve field representations and compare
+        remote_records = remote_record_obj.field_dump()
+        local_records = local_record_obj.field_dump()
+        res = compare.compare_sets(remote_records, local_records)
+
+        # Retrieve xml id representations and compare
+        # In version 11, add 'noupdate' as a field to retrieve and compare
```

Comment:
Add here the noupdate property

Reply:
I'm adding this in a follow up PR so that we can review the changes to the analysis that it causes.